src/util/dlist/dlist_debug.c

```diff
@@ -10,22 +10,17 @@
 #include <util/dlist.h>
 
 void __dlist_debug_check(const struct dlist_head *head) {
-	/* XXX: not working in util/tree because of tree_delete_link func */
-#if 0
+
+#if ENABLE_DEBUG
+
 	const struct dlist_head *p = head->prev;
 	const struct dlist_head *n = head->next;
 	uintptr_t poison = head->poison;
 
 	assert((!poison || (void *) ~poison == head) &&
```

Comment:
you can use assertf instead assert for this purpose.

Reply:
Yes, didn't know about that.